### PR TITLE
feat: add MyEnvironment page to manage per user commit images

### PR DESCRIPTION
**Pull request by agatha197**
> [!Important]
>	should be merge after https://github.com/lablup/backend.ai/pull/2010 is reflected.

- resolves https://github.com/lablup/giftbox/issues/653
- related PR: https://github.com/lablup/backend.ai/pull/2010

## Checklist
- [ ] New page: `My Environments` (> 24.03.0). If the core version is under 24.03.0 and user tries to access the `my-environment` page through the URL, it redirects to error page.
- [ ] Display the list of session committed images
- [ ] Users can delete each image by clicking the trash icon button.
- [ ] Users can filter the display columns.

## Screenshots
- list
	image

- style: border color changed
	image
	image

- style: add `cyan` color to `lablup-shields` and display constraint with Customized tag.
image

- style: use token font color to Session Name, Session ID for dark mode supporting
image

**Checklist:** (if applicable)

- [x] Mention to the original issue
- [ ] Documentation
- [x] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [x] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

File: react/src/pages/MyEnvironmentPage.tsx [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+                displayedColumnKeys?.includes(_.toString(column.key)),
+              )}
+              dataSource={processedImages as CustomizedImages[]}
+              scroll={{ x: 'max-content' }}
+            />
+            <Flex
+              justify="end"
+              style={{
+                padding: token.paddingXXS,
+              }}
```
Comment (yomybaby) [inline react/src/pages/MyEnvironmentPage.tsx:299]: If no pagination, please add `pagination={false}` to Table.

File: react/src/pages/MyEnvironmentPage.tsx [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+        <Flex direction="row" align="stretch">
+          <Popconfirm
+            title={t('dialog.ask.DoYouWantToProceed')}
+            description={t('dialog.warning.CannotBeUndone')}
+            okType="danger"
+            onConfirm={() => {
+              if (row?.id) {
+                commitForgetAndUntag({
+                  variables: {
+                    image_id: row.id,
```
Comment (yomybaby) [inline react/src/pages/MyEnvironmentPage.tsx]: Please set Popconfirm's `okText` prop to "Delete".

File: react/src/components/MainLayout/WebUISider.tsx [hunk 3]
```diff
@@ -85,6 +87,11 @@ const WebUISider: React.FC<WebUISiderProps> = (props) => {
       icon: <CloudUploadOutlined />,
       key: 'data',
     },
+    supportPendingSessionCount && {
```
Comment (yomybaby) [inline react/src/components/MainLayout/WebUISider.tsx]: `supportPendingSessionCount` is not related to user-commited image, right? It's better to add a support key for user-commited image support.

**Review by yomybaby — CHANGES_REQUESTED** [review]

File: react/src/components/MainLayout/WebUISider.tsx (hunk 3 above)
Comment (agatha197) [reply react/src/components/MainLayout/WebUISider.tsx]: Oops, I was confused with another PR. Rename completed!

File: react/src/pages/MyEnvironmentPage.tsx [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+            okText={t('button.Delete')}
+            onConfirm={() => {
+              console.log(row.key);
+              if (row.key) {
+                commitForgetAndUntag({
+                  variables: {
+                    image_id: row.key,
+                    id: row.key,
+                  },
+                  onCompleted(data) {
```
Comment (yomybaby) [inline react/src/pages/MyEnvironmentPage.tsx]: You also need to handle `errors` in `onCompleted(data, errors)`.

File: react/src/pages/MyEnvironmentPage.tsx [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+          : 'network-only',
+      fetchKey: myEnvironmentFetchKey,
+    },
+  );
+
+  const [commitForgetAndUntag, isInflightForgetAndUntag] =
+    useMutation<MyEnvironmentPageForgetAndUntagMutation>(graphql`
+      mutation MyEnvironmentPageForgetAndUntagMutation(
+        $image_id: String!
+        $id: String!
```
Comment (yomybaby) [inline react/src/pages/MyEnvironmentPage.tsx]: `$id` and `$image_id` are the same. We can use just one of them.

File: react/src/pages/MyEnvironmentPage.tsx [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+      key,
+      imageFullName,
+      namespace,
+      lang,
+      baseversion,
+      baseimage,
+      constraint,
+    };
+  });
+
```
Comment (yomybaby) [inline react/src/pages/MyEnvironmentPage.tsx]: In most cases in Relay, it's better to use GraphQL data without generating a new type. Transformation functions like `getImageFullName` can be called in React components or in the `render` function of columns.

**Review by yomybaby — CHANGES_REQUESTED** [review]

File: react/src/hooks/index.tsx [hunk 7]
```diff
@@ -178,6 +180,12 @@ interface ImageMetadata {
   }[];
 }
 
+export const getImageFullName = (image: Image | CommittedImage) => {
+  return image
+    ? `${image.registry}/${image.name}:${image.tag}@${image.architecture}`
+    : undefined;
+};
```
Comment (yomybaby) [inline react/src/hooks/index.tsx]: Let's move this function to `helper/index.tsx` because this is not a react hook.

**Comment by ironAiken2** [comment]
The sort logic in `constraint` is not working.
image

**Review by yomybaby — APPROVED** [review]
LGTM 💪

**Pull request merged**